baseapp/baseapp.go

```diff
@@ -604,6 +605,29 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+// EndBlock is an application-defined function that is called after transactions have been processed
+func (app *BaseApp) endBlock(ctx context.Context) (sdk.EndBlock, error) {
+	var endblock sdk.EndBlock
+
+	if app.endBlocker != nil {
+		eb, err := app.endBlocker(app.finalizeBlockState.ctx)
+		if err != nil {
+			panic(err)
+		}
+
+		eb.Events = sdk.MarkEventsToIndex(eb.Events, app.indexEvents)
+
+		//TODO add endblock attribute to events
+		// append the endblocker events to the finalizeblock events
+
+		endblock = eb
+	}
+
+	// TODO: Prefix with "end_blocker"
```

Comment:
I don't think this should be terribly difficult to do this in this PR. WDYT? If you want, I can wrap it up.